packages/@lwc/template-compiler/src/parser/attribute.ts

```diff
@@ -261,7 +261,7 @@ function shouldCamelCaseAttribute(element: IRElement, attrName: string) {
 }
 
 export function attributeToPropertyName(element: IRElement, attrName: string): string {
-    if (!shouldCamelCaseAttribute) {
+    if (!shouldCamelCaseAttribute(element, attrName)) {
         return attrName;
     }
     return toPropertyName(ATTRS_PROPS_TRANFORMS[attrName] || attrName);
```

Comment:
Is it just me, or should this be something like:
```suggestion
    return ATTRS_PROPS_TRANFORMS[attrName] || toPropertyName(attrName);
```

Reply:
You are right, `ATTRS_PROPS_TRANFORMS[attrName]` will always return a valid property name. No need to further transform it.